File: JavaPractice/src/FindMostRepeatedWordFromTextFile.java
import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.RandomAccessFile;
import java.util.HashMap;
import java.util.Map;
import java.util.Map.Entry;

/**
 * Given a a log file which consists of logging information like (ERROR, INFO, EXCEPTION, etc). Function
 * needs to return list of words with the frequency of occurrence. 
 * For e.g 
 * "03/22 08:51:01 INFO   :...locate_configFile: Specified configuration file: /u/user10/rsvpd1.conf
	03/22 08:51:01 INFO   :.main: Using log level 511
	03/22 08:51:01 EXCEPTION   :..settcpimage: Get TCP images rc - EDC8112I Operation not supported on socket.
	 03 
	03/22 08:51:01 ERROR   :..settcpimage: Associate with TCP/IP image name = TCPCS
	03/22 08:51:02 INFO   :..reg_process: registering process with the system
	03/22 08:51:02 ERROR   :..reg_process: attempt OS/390 registration
	03/22 08:51:02 INFO   :..reg_process: return from registration rc=0 
 * @author divyesh.shah
 *
 */
public class FindMostRepeatedWordFromTextFile {

	public static void main(String[] args) {
		// TODO Auto-generated method stub
		addDupWords();
	}
	
	public static void addDupWords() {
		
		HashMap<String, Integer> hm = new HashMap<String,Integer>();
		BufferedReader reader;
		try {
			
			File f = new File("//Users/divyesh.shah/Documents/workspace/JavaPractice/input.txt");
			reader = new BufferedReader(new FileReader(f));
			String currentline = reader.readLine();
			while(currentline != null) {
				String[] words = currentline.toLowerCase().split(" ");
				for(String word : words) {
					
					if(hm.containsKey(word)) {
						hm.put(word, hm.get(word)+1);
					}
					else {
						hm.put(word, 1);
					}
				}
				currentline = reader.readLine();
			}
			
			for(String key : hm.keySet()) {
				if(hm.get(key) > 1 ) {
					System.out.println("Key " + key + " - " + hm.get(key));
				}
			}
			
			int count = 0;
			String maxOccurence = null;
			
			for(Entry<String, Integer> e : hm.entrySet()) {
				if(e.getValue() > count) {
					maxOccurence = e.getKey();
					count = e.getValue();
				}
			}
			
			System.out.println(maxOccurence);
			
			reader.close();
			
		}
		catch (IOException e) {
			e.printStackTrace();
		}
		
	}
}

File: PythonPractice/src/FindMostRepeatedWordFromTextFile.py
import os
from collections import defaultdict

def add_dup_words(file_path):
    if not os.path.isfile(file_path):
        print(f"File not found: {file_path}")
        return

    word_count = defaultdict(int)

    try:
        with open(file_path, 'r') as file:
            for line in file:
                words = line.lower().split()
                for word in words:
                    word_count[word] += 1

        # Print words that occur more than once
        for word, count in word_count.items():
            if count > 1:
                print(f"Key {word} - {count}")

        # Find the word with the maximum occurrence
        max_occurrence_word = None
        max_count = 0

        for word, count in word_count.items():
            if count > max_count:
                max_occurrence_word = word
                max_count = count

        print(f"The word with the maximum occurrence is: {max_occurrence_word}")

    except IOError as e:
        print(f"An error occurred while reading the file: {e}")

def main():
    file_path = "/Users/divyesh.shah/Documents/workspace/JavaPractice/input.txt"
    add_dup_words(file_path)

if __name__ == "__main__":
    main()
